containers/hawkular-metrics-openshift-integration/src/main/java/org/hawkular/openshift/auth/OpenshiftAuthHandler.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2014-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.openshift.auth;
+
+
+import static io.undertow.util.Headers.ORIGIN;
+import static io.undertow.util.Methods.OPTIONS;
+
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+
+/**
+ * An Undertow handler which delegates authentication to one of the supported strategies in Openshift environments.
+ *
+ * @author Thomas Segismont
+ */
+public class OpenshiftAuthHandler implements HttpHandler {
+    private static final String OPENSHIFT_OAUTH = "openshift-oauth";
+    private static final String HTPASSWD = "htpasswd";
+    private static final String DISABLED = "disabled";
+    private static final String SECURITY_OPTION_SYSPROP = "hawkular-metrics.openshift.auth-methods";
+    private static final String SECURITY_OPTION = System.getProperty(SECURITY_OPTION_SYSPROP, OPENSHIFT_OAUTH);
+
+    private final HttpHandler containerHandler;
+    private final Authenticator authenticator;
+
+    public OpenshiftAuthHandler(HttpHandler containerHandler) {
+        this.containerHandler = containerHandler;
+        if (SECURITY_OPTION.contains(OPENSHIFT_OAUTH)) {
+            authenticator = new TokenAuthenticator(containerHandler);
+        } else if (SECURITY_OPTION.contains(HTPASSWD)) {
+            authenticator = new BasicAuthenticator(containerHandler);
```

Comment:
Ooops, right. I overlooked the "contains" method. Fixing it now.

2016-04-27 16:49 GMT+02:00 Matthew Wringe <notifications@github.com>:

> In
> containers/hawkular-metrics-openshift-integration/src/main/java/org/hawkular/openshift/auth/OpenshiftAuthHandler.java
> <https://github.com/hawkular/hawkular-metrics/pull/481#discussion_r61271587>
> :
>
> > +public class OpenshiftAuthHandler implements HttpHandler {
> > +    private static final String OPENSHIFT_OAUTH = "openshift-oauth";
> > +    private static final String HTPASSWD = "htpasswd";
> > +    private static final String DISABLED = "disabled";
> > +    private static final String SECURITY_OPTION_SYSPROP = "hawkular-metrics.openshift.auth-methods";
> > +    private static final String SECURITY_OPTION = System.getProperty(SECURITY_OPTION_SYSPROP, OPENSHIFT_OAUTH);
> > +
> > +    private final HttpHandler containerHandler;
> > +    private final Authenticator authenticator;
> > +
> > +    public OpenshiftAuthHandler(HttpHandler containerHandler) {
> > +        this.containerHandler = containerHandler;
> > +        if (SECURITY_OPTION.contains(OPENSHIFT_OAUTH)) {
> > +            authenticator = new TokenAuthenticator(containerHandler);
> > +        } else if (SECURITY_OPTION.contains(HTPASSWD)) {
> > +            authenticator = new BasicAuthenticator(containerHandler);
>
> Not exactly, that code selects which one is based on what the auth header
> is for the request and if the corresponding SECURITY_OPTION is specified.
> There is nothing in there that prevents multiple SECURITY_OPTION from being
> available.
>
> We need to support multiple SECURITY_OPTIONS, and we do use both in our
> current implementation:
> https://github.com/openshift/origin-metrics/blob/master/deployer/templates/hawkular-metrics.yaml#L75
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/hawkular/hawkular-metrics/pull/481/files/6f1f68379c3a457ff34180821423e68be5469a2b#r61271587>
>



-- 
Thomas Segismont
JBoss ON Engineering Team